### PR TITLE
Feature/aemceds-64 modal block

### PR DESCRIPTION
**Pull request by Lohit-Kumar-Sharma**
Please always provide the [Jira issue(s)] your PR is for, as well as test URLs where your change can be observed (before and after):

Fixes #AEMCEDS-64

Test URLs:
- Before: https://main--aem-competency--aem-comp.hlx.live/drafts/lohit/modal
- After: https://feature-aemceds-64-modal-block--aem-competency--aem-comp.hlx.live/drafts/lohit/modal

- [ ]  **If you are adding a new block or a variation to an existing block**, please fill below:

  Block library path: https://main--aem-competency--aem-comp.hlx.live/tools/sidekick/library.html?plugin=blocks&path=/tools/sidekick/blocks/modal&index=0

### CODE REVIEW

File: blocks/modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  width: 100vw;
+}
+
+.modal dialog .modal-content {
+  overflow-y: auto;
+  overscroll-behavior: none;
+  max-height: calc(100dvh - 60px);
+}
+
+@media (width >= 600px) {
```
Comment (mkbansal1) [inline blocks/modal/modal.css]: Correct the breakpoints, see style.css for breakpoints

File: scripts/scripts.js [hunk 2]
```diff
@@ -322,6 +334,7 @@ async function loadEager(doc) {
  * @param {Element} doc The container element
  */
 async function loadLazy(doc) {
+  autolinkModals(doc);
```
Comment (mkbansal1) [inline scripts/scripts.js]: move this method call towards end of loadLazy()

**Review by mkbansal1 — CHANGES_REQUESTED** [review]
Change the look and feel as per design
https://xd.adobe.com/view/e9937e70-1422-46de-8eb8-3dd6fd1d6025-bcc0/screen/5ded235a-7c7f-4383-8154-5d1beca81401

X icon is different

File: blocks/modal/modal.css (hunk 1 above)
Comment (Lohit-Kumar-Sharma) [reply blocks/modal/modal.css]: This is addressed, corrected the breakpoints to 1280px

File: scripts/scripts.js (hunk 2 above)
Comment (Lohit-Kumar-Sharma) [reply scripts/scripts.js]: This is addressed.

**Comment by Lohit-Kumar-Sharma** [comment]
> Change the look and feel as per design https://xd.adobe.com/view/e9937e70-1422-46de-8eb8-3dd6fd1d6025-bcc0/screen/5ded235a-7c7f-4383-8154-5d1beca81401
> 
> X icon is different

Lohit - This is addressed

**Review by mkbansal1 — APPROVED** [review]

**Pull request merged**